### ISSUE TITLE
Import of Typescript interface is not recognised correctly

### ISSUE THREAD

**Issue by majelbstoat** [benmosher/eslint-plugin-import#1120]
Opening per @ljharb. A similar issue was reported in https://github.com/benmosher/eslint-plugin-import/issues/726, but this is for Typescript interface imports.

Defined in react-select:

```ts
export interface Option<TValue = OptionValues> { // ... }
```

Imported:
```ts
import Select, { Option } from 'react-select'
```

Error:
```
[eslint] Option not found in 'react-select' (import/named)
```

Relevant packages:
```
    "eslint": "4.19.1",
    "eslint-plugin-import": "2.12.0",
    "typescript": "2.9.1",
    "typescript-eslint-parser": "16.0.0"
```

**Comment by ljharb** [comment]
Historically eslint (and thus, eslint plugins) have only supported Flow, not TypeScript.

I think adding typescript support is a pretty massive effort, but if it can be done easily, that sounds fine.

**Comment by mockdeep** [comment]
Just ran into this, too:

```ts
import moment, {Moment} from 'moment';
```

error:

```
1:17  error  Moment not found in 'moment'             import/named
```

**Comment by ljharb** [comment]
In that case, moment is CJS, so it only has a default export.

**Comment by mockdeep** [comment]
TypeScript also pulls in the type information from it's `.d.ts` file:

**Comment by Jessidhia** [comment]
https://github.com/benmosher/eslint-plugin-import/issues/1282 is nearly a duplicate of this, but this issue also includes types provides through external `.d.ts` files.

I'm not sure if it is possible to deal with this 100% correctly without a full type checker pass; a possible workaround is to check if there is a `.d.ts` available for parsing its imports instead before checking other extensions, but that won't work with `declare module`-style declarations (which can declare any module name they want if they're in ambient context).

`@typescript-eslint/eslint-plugin` will either need its own rule for this, or `eslint-plugin-import` should be taught to make use of the type services from `@typescript-eslint/parser` when they are available.

**Comment by Primajin** [comment]
After upgrading to `2.17.1` this is fixed for me 🎉

**Comment by ljharb** [comment]
Awesome, closing (happy to reopen if people are still experiencing it)

**Comment by alesn** [comment]
Unfortunately, this issue still exists in 2.17.2 (using @typescript-eslint/parser as recommended in the Typescript section of eslint-plugin-import Readme [not the now deprecated typescript-eslint-parser mentioned in the configuration above])

**Comment by kachkaev** [comment]
@alesn would you mind opening a new issue and shading a minimal reproduction?

**Comment by alesn** [comment]
Has this issue been resolved for anyone who uses @typescript-eslint/parser, which should now be used?

**Comment by Primajin** [comment]
I will double check that and let you know

**Comment by Primajin** [comment]
@alesn Sorry I forgot to answer you earlier.

Yes I am running `"@typescript-eslint/parser": "^1.6.0"` and `"eslint-plugin-import": "^2.17.1"`.
There is also `2.17.2` out in the meantime.

In .eslintrc.js I have

```js
parser: '@typescript-eslint/parser',
overrides: {
  files: ['**/*.ts', '**/*.tsx'],
  parser: '@typescript-eslint/parser',
  rules: {
    'import/namespace': 'off',
    'import/no-duplicates': 'off',
    'import/no-unresolved': 'off',
    'no-restricted-globals': 'off',
    'no-undef': 'off'
  }
},
```

Hope that helps.

**Comment by cloakedninjas** [comment]
For anyone arriving from Google - you can work around this with a tip from [#1699 ](https://github.com/benmosher/eslint-plugin-import/issues/1699#issuecomment-604246968)

> Although now that import type exists, I'd suggest using that.

```js
import type { unitOfTime } from 'moment';
